### ISSUE TITLE
File URL option for the GenAI File API 

### ISSUE THREAD

**Issue by TecEash1** [google-gemini/generative-ai-js#130]
At the moment a file path has to be utilised: ``mimeType: mime.lookup(filePath),``. However it would be helpful if a file URL could also be used instead of a local file path: ``mimeType: mime.lookup(rawfileURL),``

**Comment by TecEash1** [comment]
That is correct, sorry about saying ``mime.lookup``, what you mentioned about using the ``GoogleAIFileManager.uploadFile`` method with a fileURL is what I meant to say. 😅

**Comment by lahirumaramba** [comment]
Hey @TecEash1 just to clarify, by `file URL` do you mean a url that starts with the `file://` protocol?

**Comment by TecEash1** [comment]
sorry, thats not what I initially meant. I meant a raw file starting with the ``https``/``http`` protocol. Such as:

https://www.google.com/images/branding/googlelogo/2x/googlelogo_light_color_272x92dp.png

**Comment by mynamebrody** [comment]
+1. We, at Zapier, would love the ability to pass in a file via the https/http protocol

**Comment by EmilyYond** [comment]
+1 to be able to send image/video urls instead of fetching them to resend them to you would be brilliant!

**Comment by jin-ds** [comment]
+1 here as well—thank you!

**Comment by hsubox76** [comment]
Unfortunately this would be beyond a simple SDK fix and would require a change to the Gemini API.

If we provided this functionality in the SDK, we'd be doing the same thing as you would - downloading the image and uploading it somewhere, but doing it invisibly, under the hood, which I don't think provides that much value and introduces more room for error.

I will pass along to the Gemini team that users are interested in this feature, or you may get more visibility posting in the Gemini API discussion forum: https://discuss.ai.google.dev/

**Comment by hsubox76** [comment]
Also, can everyone use the "thumbs up" emote on the first post to indicate "+1" on interest? It will be a handy way for us to see the total interest in this feature request and avoid making this issue thread too long.

**Comment by ChrisCates** [comment]
@hsubox76, this issue got lost and never addressed.

Instead of refactoring platform services (more work for marginal returns).
Add a patch to the client side easier and thus less friction.

Add a new method for both python and sdk:

```python
file = await genai.remote_file('...')
```

```typescript
const file: FileMetadataResponse = await genai.remoteFile('...')
```

On the client, it does the following:

1. Downloads the file to a local `tmp` path
2. Uploads the file to the Google Filesystem

I think this methodology and implementation will reduce a lot of friction of being able to produce a PR that everyone will benefit from.

**Comment by BrianHung** [comment]
https://github.com/sindresorhus/file-type covers most use cases, would be great if it was automatic though.

**Comment by shane-cw** [comment]
Yes, please support passing in an image via http url like the other LLM providers do. It's awkward to have to upload via File API first. We have too many images to pass via base64 string.

Bump to:

> +1. We, at Zapier, would love the ability to pass in a file via the https/http protocol. Especially with the new ability to upload Videos. If we could just pass a URL to a video instead of having to download it ourselves, just to re-package it up and upload it to you, if we could just pass along the same URL that we have access too, that would be amazing.

**Comment by code-soubhik** [comment]
Hello everyone,
I have raise a PR #413 

It includes the changes of uploading files remotely

@IvanLH Let me know if we need to modify anything.
